ITensorGaussianMPS/src/gmps.jl

```diff
@@ -181,82 +171,61 @@ function quadratic_hamiltonian(os::OpSum)
     sites[n] = ntuple(n -> ITensors.site(term[n]), Val(2))
     nsites = max(nsites, maximum(sites[n]))
   end
-  quadrants_present = unique(quads)
-  # currently compatible formats:
-  # [(2,2)] -> hopping only
-  # [(1,2),(2,2)] -> hopping and pairing
-  @show quadrants_present
   ElT = all(isreal(coefs)) ? Float64 : ComplexF64
```

Comment:
Can we use:
```julia
coef_type = reduce(Base.promote_typeof, coefs)
ElT = all(isreal(coefs)) ? real(coef_type) : coef_type
```
to make it more agnostic about the precision (i.e. more gracefully handle `Float32`, etc.)?

Reply:
The first line returns any for me on Julia v1.8. I wrote an alternative but it's much less elegant than this solution.

Reply:
I pushed my fix to this, it's a bit more verbose.